src/server/scripts/Spells/spell_mage.cpp

```diff
@@ -450,29 +450,27 @@ class spell_mage_cold_snap : public SpellScript
     {
         return ValidateSpellInfo
         ({
-            SPELL_MAGE_CONE_OF_COLD,
-            SPELL_MAGE_FROST_NOVA,
+            SPELL_MAGE_ICE_BLOCK,
             SPELL_MAGE_ICE_BARRIER,
-            SPELL_MAGE_ICE_BLOCK
+            SPELL_MAGE_CONE_OF_COLD,
+            SPELL_MAGE_FROST_NOVA
         });
     }
 
     void HandleDummy(SpellEffIndex /*effIndex*/)
     {
-        GetCaster()->GetSpellHistory()->ResetCooldowns([](SpellHistory::CooldownStorageType::iterator itr)
+        std::vector<uint32> spellsToReset =
```

Comment:
you are creating a new vector for each cast but the list never changes, so we can make it constant;
use
```c
static uint32 constexpr SpellsToReset[] = {
    SPELL_MAGE_ICE_BLOCK,
    SPELL_MAGE_ICE_BARRIER,
    SPELL_MAGE_CONE_OF_COLD
};
```
outside the function, like its done in `spell_rog_roll_the_bones`.
Usage doesnt change btw